app/assets/javascripts/contacts.js

```diff
@@ -124,28 +150,32 @@ var updateContact = function() {
     notes: $("[data-role='notes']").val()
   };
 
-  var id = $("[data-role='id']").val();
-  var method = "POST";
-  var updateUrl = url;
-
-  if (id > 0) {
-    method = "PATCH";
-    updateUrl = updateUrl + "/" + id;
+  var validContact = validateContact(contact);
+
+  if (validContact) {
+    var id = $("[data-role='id']").val();
+    var method = "POST";
+    var updateUrl = url;
+
+    if (id > 0) {
```

Comment:
What is this check for? Maybe pull it out into a function or a variable to name what it is to make the code more clear.